src/core/DatetimeClasses.ts

```diff
@@ -0,0 +1,382 @@
+import {
+    getOffsetSecondsFromTimezoneString,
+    getTimezoneOffsetFromSeconds
+} from '../util/DatetimeUtil';
+
+/**
+ * ### Local time object
+ * * Represents time in day without timezone.
+ */
+export class HzLocalTime {
+    private static readonly timeStringRegex = /(\d\d):(\d\d):(\d\d)(\.\d+)?/;
+
+    /**
+     * @param hour The hour-of-day to represent, from 0 to 23
+     * @param minute The minute-of-hour to represent, from 0 to 59
+     * @param second The second-of-minute to represent, from 0 to 59
+     * @param nano The nano-of-second to represent, from 0 to 999,999,999
+     * @throws RangeError if any of the arguments are invalid
+     */
+    constructor(
+        readonly hour: number,
+        readonly minute: number,
+        readonly second: number,
+        readonly nano: number
+    ) {
+        if (!Number.isInteger(hour) || !Number.isInteger(minute) || !Number.isInteger(second) || !Number.isInteger(nano)) {
+            throw new RangeError('Illegal arguments given to HzLocalTime. All arguments must be integers.');
+        }
+        if (!(hour >= 0 && hour <= 23)) {
+            throw new RangeError('Hour-of-day must be between 0-23');
+        }
+        if (!(minute >= 0 && minute <= 59)) {
+            throw new RangeError('Minute-of-hour must be between 0-59');
+        }
+        if (!(second >= 0 && second <= 59)) {
+            throw new RangeError('Second-of-minute must be between 0-59');
+        }
+        if (!(nano >= 0 && nano <= 999_999_999)) {
+            throw new RangeError('Nano-of-second must be between 0-999_999_999');
+        }
+    }
+
+    /**
+     * Constructs a new {@link HzLocalTime} object from timeString.
+     * @param timeString A string in the form HH:mm:ss.SSS, where the last part represents nanoseconds and optional.
+     * At most 9 digits allowed for nanosecond value. If more than 9 digits are given, the first 9 of them are used.
+     * @throws RangeError if invalid timeString is given
+     */
+    static fromString(timeString: string): HzLocalTime {
+        if (typeof timeString !== 'string') {
+            throw new RangeError('String expected.');
+        }
+        let match = timeString.match(HzLocalTime.timeStringRegex);
+        if (!match) {
+            throw new RangeError('Illegal time string. Expected a string in HH:mm:ss.SSS format');
+        }
+        match = match.slice(1); // Discard first match which is the first complete match
```

Comment:
Instead of slicing, we can use it like match[1], match[2], match[3], match[4] right? We do it like this in LocalDate

Reply:
yes we can. on it